js/src/lib/components/InferenceWidget/shared/WidgetExample.ts

```diff
@@ -108,3 +108,7 @@ export type WidgetExample<TOutput = WidgetExampleOutput> =
 	| WidgetExampleTableDataInput<TOutput>
 	| WidgetExampleZeroShotTextInput<TOutput>
 	| WidgetExampleSentenceSimilarityInput<TOutput>;
+
+type KeysOfUnion<T> = T extends any ? keyof T : never;
```

Comment:
Isn't this just `keyof T`?

Reply:
they are different. In `keyof T` case, ts errs